src/api/imageProxy.ts

```diff
@@ -0,0 +1,15 @@
+import { ImageProxyResponse } from '../components/editor/markdown-renderer/replace-components/image/types'
+import { expectResponseCode, getBackendUrl } from '../utils/apiUtils'
+import { defaultFetchConfig } from './default'
+
+export const getProxiedUrl = async (imageUrl: string): Promise<ImageProxyResponse> => {
```

Comment:
Why are we using a call that delivers an image URL? Why can't we send the image url to the backend and receive an image?

Reply:
Because the image proxy is a separate piece of software that stands besides the backend.

One could implement it in the backend, but then you have to deal with mime-types, images sizes, dead URL, etc. in our backend code and that is just a unneeded hassle. Or we start to pipe things through our backend which is something `node` is not meant to do, due to its single-threaded, event-queue-based system.

Reply:
That makes sense. I guess it's okay.

Reply:
Did you think about the possibility to set the proxy url into the frontend config and sending requests directly to the proxy?

Reply:
At least camo uses a secret that is shared between a backend and the proxy in order to not being a fully open proxy in order to prevent abuse and apply rate-limiting  on application side. That's at least my idea why the backend should be involved here.

See: https://github.com/cactus/go-camo#how-it-works

Reply:
Hm okay. How do we prevent that the backend endpoint /media/proxy becomes a ffa proxy?

Reply:
CSRF?

Reply:
Or:

Is there any plan to protect this endpoint from abuse with non-hedgedoc use?

Reply:
Well. I think that doesn't matter here. That's more of a backend story.